### ISSUE TITLE
Recompilation does not happen for Typescript after write-file-webpack-plugin 4.2.0

### ISSUE THREAD

**Issue by changLiuUNSW** [gajus/write-file-webpack-plugin#63]
I am using TS loader + write-file-webpack-plugin

If there is an error in the code eg: typo, the errror will show and it does not trigger recompilation after the fix.

This issue does not happen for 4.2.0 and below. I think there is some code introduced after 4.2.0 break the recompilation

**Comment by viktorlarsson** [comment]
I am experiencing the same issue. Same occurs when using sass-loader

**Comment by changLiuUNSW** [comment]
I have done the research and I believe I find the reason. 
The plugin used to output assets has been moved to ```after-emit``` from ```done``` after version 4.2.0 in order to support Webpack 4+.
By design, ```after-emit``` will stop recompilation  if a plugin calls callback with error:
https://github.com/webpack/webpack/issues/4326
Here is the code that cause the issue:
```
      if (options.exitOnErrors && compilation.errors.length) {
        // eslint-disable-next-line promise/prefer-await-to-callbacks
        callback(compilation.errors); //<---- Here 

        return;
      }
```
I have noticed that the flag ```exitOnErrors```  is ```true``` by default. 
I believe the author original intention is to stop writing files on webpack errors, but now it will also stop recompilation. 
As a workaround for now, we need to change it to ```false``` like: 
```
new WriteFilePlugin({
      exitOnErrors : false
})
```
@gajus 
I have raised a PR to fix this issue
https://github.com/gajus/write-file-webpack-plugin/pull/64
